File: app/src/main/java/com/mycompany/myapp/app/AndroidModule.java
package com.mycompany.myapp.app;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidModule {
    private final MainApplication application;

    public AndroidModule(MainApplication application) {
        this.application = application;
    }

    @Singleton
    @Provides
    public Context provideContext() {
        return application;
    }

    @Singleton
    @Provides
    public Application provideApplication() {
        return application;
    }
}

File: app/src/main/java/com/atomicrobot/carbon/app/AndroidModule.kt
package com.atomicrobot.carbon.app

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AndroidModule(private val application: MainApplication) {
    @Singleton
    @Provides
    fun provideContext(): Context  = application

    @Singleton
    @Provides
    fun provideApplication(): Application = application
}
